src/App.js

```diff
@@ -59,6 +59,8 @@ let App = () => {
   )
   useEffect(() => localStorage.timeZone = timeZone, [timeZone])
 
+  findPreviousState()
```

Comment:
Maybe loadCachedQSIfNotExists()? Bit verbose ik, honestly probs easier to just put the whole function here

Reply:
I sort of wanted to keep the URL functionality close to each other, but I'll change the function name.